### PR TITLE
[DO NOT MERGE] Neworigins

### PR DESCRIPTION
**Pull request by artyomtrityak**

### CODE REVIEW

File: army.cpp [hunk 1]
```diff
@@ -843,8 +1012,12 @@ void Army::Win(Battle * b,ItemList * spoils)
 	WriteLosses(b);
 
 	int na = NumAlive();
+	int loses_percent = 100;
+	if (na != 0 && count != 0) {
+		loses_percent = 100 - ((na * 100) / count);
+	}
 
-	if (count - na) wintype = WIN_DEAD;
+	if (loses_percent >= 5) wintype = WIN_DEAD;
```
Comment (artyomtrityak) [inline army.cpp:1020]: TODO: add rules

File: army.cpp [hunk 2]
```diff
@@ -967,6 +1140,22 @@ int Army::NumFront()
 	return (canfront + notfront - canbehind);
 }
 
+int Army::NumBehind() {
```
Comment (artyomtrityak) [inline army.cpp:1143]: TODO: add documentation or remove it

File: army.cpp [hunk 3]
```diff
@@ -1113,10 +1312,31 @@ int Army::RemoveEffects(int num, char const *effect)
 	return(ret);
 }
 
-int Army::DoAnAttack(char const *special, int numAttacks, int attackType,
+WeaponBonusMalus* GetWeaponBonusMalus(WeaponType *weapon, WeaponType *target) {
```
Comment (artyomtrityak) [inline army.cpp:1315]: TODO add documentation or remove it

File: basic/rules.cpp [hunk 4]
```diff
@@ -228,7 +228,17 @@ static GameDefs g = {
...
 	0, //DISPERSE_GATE_NUMBERS
-	0	// UNDEATH_CONTAGION
+	0,	// UNDEATH_CONTAGION
+	0,  // REGIONS_ECONOMY
+	0, // ADVANCED_TACTICS
+	0,	// OVERWHELMING
+	0, // STRICT_GUARD
+	1, // OCEAN_GUARD
+	BattleLogLevel::NORMAL,	// BATTLE_LOG_LEVEL
+	0, // EXTENDED_FORT_DEFENCE
```
Comment (artyomtrityak) [inline basic/rules.cpp:238]: Add documentation

File: basic/rules.cpp [hunk 5]
```diff
@@ -228,7 +228,17 @@ static GameDefs g = {
 	0, // ALLIES_NOAID
 	0, // HARDER_ASSASSINATION
 	0, //DISPERSE_GATE_NUMBERS
-	0	// UNDEATH_CONTAGION
+	0,	// UNDEATH_CONTAGION
+	0,  // REGIONS_ECONOMY
+	0, // ADVANCED_TACTICS
+	0,	// OVERWHELMING
+	0, // STRICT_GUARD
```
Comment (artyomtrityak) [inline basic/rules.cpp:235]: Add documentation

File: basic/rules.cpp [hunk 6]
```diff
@@ -228,7 +228,17 @@ static GameDefs g = {
...
+	1, // OCEAN_GUARD
+	BattleLogLevel::NORMAL,	// BATTLE_LOG_LEVEL
+	0, // EXTENDED_FORT_DEFENCE
+	0, // SPOILS_NO_TRADE
+	1,	// WORLD_EVENTS
```
Comment (artyomtrityak) [inline basic/rules.cpp:240]: Improve world events or remove them

File: battle.cpp [hunk 7]
```diff
@@ -565,29 +840,40 @@ void Game::GetSides(ARegion *r, AList &afacs, AList &dfacs, AList &atts,
...
+					((Object *) elem)->capacity = 0;
+					((Object *) elem)->shipno = ((Object *) elem)->ships.Num();
+					continue;
+				}
+
 				/* Set building capacity */
 				if (o->incomplete < 1 && o->IsBuilding()) {
 					o->capacity = ObjectDefs[o->type].protect;
 					o->shipno = 0;
+					// AddLine("Fortification bonus added for ", o->name);
```
Comment (artyomtrityak) [inline battle.cpp:868]: remove

File: items.cpp [hunk 8]
```diff
@@ -535,6 +572,25 @@ static AString MonResist(int type, int val, int full)
 	return temp;
 }
 
+static AString FMIResist(int type, int val, int full)
+{
+	AString temp = "This FMI ";
```
Comment (artyomtrityak) [inline items.cpp:577]: explain FMI

**Comment by valdisz** [comment]
@artyomtrityak this can be closed because another PR (#142) exists from more fresh branch

**Pull request closed without merging**